test/common/router/router_test.cc

```diff
@@ -1157,65 +1157,16 @@ TEST_F(RouterTest, EnvoyAttemptCountInResponseWithRetries) {
   EXPECT_TRUE(verifyHostUpstreamStats(1, 1));
 }
 
-// Append cluster with default header name.
```

Comment:
Why are all these tests removed? Shouldn't there be new tests that cover the local reply path?

Reply:
These tests are used to test the features of DebugConfig. (Note the DebugConfig is not exposed publicly )


But this PR removed these features because it could be covered by our public response_headers_to_add.